lib/monocle.js

```diff
@@ -5,20 +5,27 @@ var ResourceCache = require('./cache/resource');
 var querystring = require('querystring');
 var CollectionCache = require('./cache/collection');
 
-function Monocle(http) {
+function Monocle(http, promise) {
     this._http = http;
     this._base = '/';
     this._cache = new Store(new ResourceCache('monocle', { capacity: 100 }));
     this._batched = [];
     this._batchTimeout = null;
     this._queuedGets = {};
+    this._promise = Promise || promise;
```

Comment:
I think this should be the other way around. If you inject `promise`, it should replace the default promise implementation.